pyproject.toml

```diff
@@ -0,0 +1,369 @@
+[build-system]
+requires = ["setuptools~=68.0.0", "wheel~=0.40.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "aiosupervisor"
+version = "0.1.0"
+license = { text = "Apache-2.0" }
+description = "Asynchronous python client for Home Assistant Supervisor."
+readme = "README.md"
+authors = [
+    { name = "The Home Assistant Authors", email = "hello@home-assistant.io" },
+]
+keywords = ["docker", "home-assistant", "api", "client-library"]
+requires-python = ">=3.12.0"
+dependencies = [
```

Comment:
There's the `project.optional-dependencies` section we could add and have a `dev` list with the dev dependencies that are all pinned.

Then we don't need `requirements_dev.txt`, I think.

Reply:
Does that mean I would need to introduce a toml parser here:
https://github.com/home-assistant-libs/python-supervisor-client/blob/b0eb227081859b4e5d2c6fe06d8e444cc02097a5/.github/workflows/ci.yaml#L50

And here:
https://github.com/home-assistant-libs/python-supervisor-client/blob/b0eb227081859b4e5d2c6fe06d8e444cc02097a5/script/devcontainer_bootstrap#L6

To set up environments in those places? Or is there a way to tell pip to install it as dev and set up the dependencies listed there accordingly?

Reply:
Yes, each list name is an "extra" set of dependencies and can be installed with eg `pip install somelibrary[extraname]`.

Reply:
@MartinHjelmare I had borrowed the venv caching from supervisor throughout the github actions. But how it works is by including a hash of the requirements as a key as you can see here:
https://github.com/home-assistant-libs/python-supervisor-client/blob/b0eb227081859b4e5d2c6fe06d8e444cc02097a5/.github/workflows/ci.yaml#L41-L43

Is there a good way to replicate this behavior without the separate requirements file? Perhaps just hash the `pyproject.toml` file itself?

Reply:
Yes, use that file.